numerical_methods/false_position.cpp

```diff
@@ -13,62 +18,108 @@
  * continued till a close enough approximation is achieved.
  *
  * \see newton_raphson_method.cpp, bisection_method.cpp
+ *
+ * \author Unknown author
+ * \author [Samruddha Patil](https://github.com/sampatil578)
  */
-#include <cmath>
-#include <cstdlib>
-#include <iostream>
-#include <limits>
+#include <cmath>      /// for math operations
+#include <iostream>   /// for io operations
 
-#define EPSILON \
-    1e-6  // std::numeric_limits<double>::epsilon()  ///< system accuracy limit
-#define MAX_ITERATIONS 50000  ///< Maximum number of iterations to check
-
-/** define \f$f(x)\f$ to find root for
+/**
+ * @namespace numerical_methods
+ * @brief Numerical methods
  */
-static double eq(double i) {
-    return (std::pow(i, 3) - (4 * i) - 9);  // origial equation
+namespace numerical_methods {
+/**
+ * @namespace false_position
+ * @brief Functions for [False Position]
+ * (https://en.wikipedia.org/wiki/Regula_falsi) method.
+ */
+namespace false_position {
+/**
+* @brief This function gives the value of f(x) for given x.
+* @param x value for which we have to find value of f(x).
+* @return value of f(x) for given x.
+*/  
+static double eq(double x) {
+    return (x*x-x);  // original equation
 }
 
-/** get the sign of any given number */
-template <typename T>
-int sgn(T val) {
-    return (T(0) < val) - (val < T(0));
+/**
+* @brief This function finds root of the equation in given interval i.e. (x1,x2).
+* @param x1,x2 values for an interval in which root is present.
+  @param y1,y2 values of function at x1, x2 espectively.
+* @return root of the equation in the given interval.
+*/  
+static double regula_falsi(double x1,double x2,double y1,double y2){
+    double diff = x1-x2;
+    if(diff<0){
+        diff= (-1)*diff;
+    }
+    if(diff<0.00001){          
+        if(y1<0)
+        y1=-y1;
+        if(y2<0)
+        y2=-y2;
+        if(y1<y2)
+        return x1;
+        else
+        return x2;
+    }
+    double x3=0,y3=0;
+    x3 = x1 - (x1-x2)*(y1)/(y1-y2);
+    y3 = eq(x3);
+    return regula_falsi(x2,x3,y2,y3);
 }
 
-/** main function */
-int main() {
-    double a = -1, b = 1, x, z, m, n, c;
-    int i;
-
-    // loop to find initial intervals a, b
-    for (int i = 0; i < MAX_ITERATIONS; i++) {
-        z = eq(a);
-        x = eq(b);
-        if (sgn(z) == sgn(x)) {  // same signs, increase interval
-            b++;
-            a--;
-        } else {  // if opposite signs, we got our interval
-            break;
-        }
+/**
+* @brief This function prints roots of the equation.
+* @param root which we have to print. 
+* @param count which is count of the root in an interval [-range,range].
+*/  
+void printRoot(double root,int count){
```

Comment:
It is recommended to use `uint64_t` rather than `int`.
These functions can be made a `const`, as they aren't being modified.
```suggestion
void printRoot(const double &root, const uint64_t &count){
```

Reply:
Variables root and count are being modified in main function. So, we can't use const .
Program is not showing any response on changing int by uint64_t.

Reply:
Please change the `int` variables to `uint64_t` (or their appropriate size: `uint8_t`, `uint16_t`, etc.).
The `count` variable can be made a `const` and passed by reference; tested on Gitpod and runs fine.